testsuite/forge/tests/fire.rs

```diff
@@ -0,0 +1,160 @@
+use diem_sdk::{
+    client::BlockingClient,
+    transaction_builder::{Currency, TransactionFactory},
+    types::{chain_id::ChainId, AccountKey, LocalAccount},
+};
+use diem_swarm::swarm::DiemSwarm;
+use forge::{forge_main, ForgeConfig, Result, *};
+
+fn main() -> Result<()> {
+    let tests = ForgeConfig {
+        public_usage_tests: &[&FundAccount],
+        admin_tests: &[&GetMetadata],
+    };
+
+    forge_main(tests, LocalFactory::new())
+}
+
+impl Swarm for LocalSwarm {
+    fn admin_info(&mut self) -> AdminInfo<'_> {
+        AdminInfo::new(
+            &mut self.root_account,
+            &mut self.treasury_compliance_account,
+            &self.url,
+            ChainId::test(),
+        )
+    }
+
+    fn public_info(&mut self) -> PublicInfo<'_> {
+        PublicInfo::new(
+            &self.url,
+            ChainId::test(),
+            Coffer::TreasuryCompliance {
+                transaction_factory: TransactionFactory::new(ChainId::test()),
```

Comment:
Maybe add information like the `ChainId` to the `Swarm` as a separate function?  Then you can ensure not making a bunch of mistakes changing it 3 places?